process/process_posix.go

```diff
@@ -80,32 +80,31 @@ func PidExistsWithContext(ctx context.Context, pid int32) (bool, error) {
 		return false, err
 	}
 
-	if ! strings.HasPrefix(common.HostProc(),"/proc") { //Means that proc is mounted not to /proc, and we highly
-		// likely running inside container with a different process namespace (by default), so we check pid existence
-		// based on /<HOST_PROC>/proc/<PID> folder
-		file, err := os.Open(common.HostProc(strconv.Itoa(int(pid))))
-		defer file.Close()
-		if err!=nil{return false, err} else{return true, nil}
-	}else{ //Assume that we are in the same process namespace, try to signal process
+	if _, err := os.Stat(common.HostProc()); err == nil { //Means that proc filesystem exist
```

Comment:
Currently it is 2 calls to os.Stat, can be reduced to 1 if we cache somewhere in global var the result of this line.
@Lomanic, @shirou could you please advise the right place for that?

Reply:
These `os.Stat` target are different and second one is a fallback. so we can not merge.

Reply:
Well, they are. But I feel that  my point is not clear. The os.Stat(common.HostProc()) is useless to call every time when we check pid existence, this is something that could be checked once at startup or init. Possible use case here is when you have an array of PIDs and you check in cycle if the PID still exist. In this case you have n=len(pidArray) -1 useless os.Stat calls.